crates/sui-types/src/messages.rs

```diff
@@ -291,9 +299,31 @@ impl MoveCall {
                 ),
             })
             .flatten()
-            .chain([InputObjectKind::MovePackage(*package)])
+            .chain(packages.into_iter().map(InputObjectKind::MovePackage))
             .collect()
     }
+
+    fn type_argument_packages(packages: &mut BTreeSet<ObjectID>, type_argument: &TypeTag) {
```

Comment:
nit: This could be a top-level function, and probably the name should hint that it's accumulating the package names into `packages`, how about `add_type_argument_packages`?